internal/validation/validation.go

```diff
@@ -1,38 +1,17 @@
 package validation
 
 import (
-	"fmt"
-	"net/url"
-)
-
-// IsURLWithNoFragment is a SchemaValidateFunc which tests if the provided value
-// is of type string and a valid URL with no fragment.
-func IsURLWithNoFragment(i interface{}, k string) (warnings []string, errors []error) {
-	v, ok := i.(string)
-	if !ok {
-		errors = append(errors, fmt.Errorf("expected type of %q to be string", k))
-		return
-	}
-
-	if v == "" {
-		errors = append(errors, fmt.Errorf("expected %q url to not be empty, got %v", k, i))
-		return
-	}
+	"strings"
 
-	u, err := url.Parse(v)
-	if err != nil {
-		errors = append(errors, fmt.Errorf("expected %q to be a valid url, got %v: %+v", k, v, err))
-		return
-	}
-
-	if u.Host == "" {
-		errors = append(errors, fmt.Errorf("expected %q to have a host, got %v", k, v))
-		return
-	}
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
 
-	if u.Fragment != "" {
-		errors = append(errors, fmt.Errorf("expected %q to have a url with an empty fragment. %s", k, v))
+func IsURLWithHTTPSorEmptyString(i interface{}, s string) ([]string, []error) {
+	_, errors := validation.IsURLWithHTTPS(i, s)
+	for _, err := range errors {
+		if !strings.Contains(err.Error(), "url to not be empty") {
```

Comment:
Can we check this some other way, instead of looking for a hardcoded error string? Maybe checking for empty string before the `IsURLWithHTTPS` validation?

Reply:
Wondering the same thing. Although that seems like too easy of a solution so wondering what we're missing.

Reply:
I wanted to avoid duplicating logic from within the IsURLWithHTTPS, specifically checking for the string type and then returning nil on an empty string. But I can skip using IsURLWithHTTPS and implement our own logic that's a tiny itsy bitsy bit more performant.

Reply:
Can't we keep `IsURLWithHTTPS` and do an empty string check before? No need to remove it. Note that it's not about performance, it's about depending on the error message string

Reply:
Though it's not a blocker for me.

Reply:
I updated this in [`1007111` (#451)](https://github.com/auth0/terraform-provider-auth0/pull/451/commits/10071111dbfbc5059db48f3de7939aff7b6da2d5) without making use of the `IsURLWithHTTPS` func from the SDK. I personally didn't like duplicating the type check and returning immediately if the url is empty as there was another check inside the sdk func for empty urls.